extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -585,35 +650,31 @@ public String getHlsUrl() throws ParsingException {
     @Nonnull
     public List<SubtitlesStream> getSubtitles(final MediaFormat format) throws ParsingException {
         assertPageFetched();
-        // If the video is age-restricted getSubtitles will fail
-        if (getAgeLimit() != NO_AGE_LIMIT) {
-            return Collections.emptyList();
-        }
-        if (subtitles != null) {
-            // Already calculated
-            return subtitles;
-        }
 
-        final JsonObject renderer = playerResponse.getObject("captions")
-                .getObject("playerCaptionsTracklistRenderer");
-        final JsonArray captionsArray = renderer.getArray("captionTracks");
-        // TODO: use this to apply auto translation to different language from a source language
-        // final JsonArray autoCaptionsArray = renderer.getArray("translationLanguages");
-
-        subtitles = new ArrayList<>();
-        for (int i = 0; i < captionsArray.size(); i++) {
-            final String languageCode = captionsArray.getObject(i).getString("languageCode");
-            final String baseUrl = captionsArray.getObject(i).getString("baseUrl");
-            final String vssId = captionsArray.getObject(i).getString("vssId");
-
-            if (languageCode != null && baseUrl != null && vssId != null) {
-                final boolean isAutoGenerated = vssId.startsWith("a.");
-                final String cleanUrl = baseUrl
-                        .replaceAll("&fmt=[^&]*", "") // Remove preexisting format if exists
-                        .replaceAll("&tlang=[^&]*", ""); // Remove translation language
-
-                subtitles.add(new SubtitlesStream(format, languageCode,
-                        cleanUrl + "&fmt=" + format.getSuffix(), isAutoGenerated));
+        if (subtitles == null) {
+            subtitles = new ArrayList<>();
+            final JsonObject renderer = playerResponse.getObject("captions")
+                    .getObject("playerCaptionsTracklistRenderer");
+            final JsonArray captionsArray = renderer.getArray("captionTracks");
+            // TODO: use this to apply auto translation to different language from a source language
+            // final JsonArray autoCaptionsArray = renderer.getArray("translationLanguages");
```

Comment:
###### Note: TeamNewPipe/NewPipe#5338 is related

Reply:
So should I change something?

Reply:
No